python/runtime/tensorflow/predict.py

```diff
@@ -108,8 +108,11 @@ def eval_input_fn(batch_size, cache=False):
     column_names.append(result_col_name)
 
     column_names.extend(extra_result_cols)
+
     with db.buffered_db_writer(conn, result_table, column_names, 100) as w:
-        for features in pred_dataset:
+        for features, (row, _) in zip(pred_dataset, predict_generator()):
```

Comment:
I wonder if the result order of the `SELECT ...` statement is always deterministic? I mean: does the data order of `pred_dataset` can always be the same as `predict_generator`?

Reply:
`dataset` is generated by `generator`, from the source code of [from_generator](https://github.com/tensorflow/tensorflow/blob/v2.3.1/tensorflow/python/data/ops/dataset_ops.py#L823), it stays the same order expect calling [shuffle](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#shuffle)